### PR TITLE
Implement half-Student's T as a first-class distribution

### PR DESCRIPTION
**Pull request by aloctavodia**
* Implement HalfStudentT like the rest of the distributions (better docs, LaTeX representation, random samples, etc)
* Finally remove StudentTpos (superseeded by HalfStudentT) and remove deprecation warning (it was introduced more than one year ago)
* (Partially) fix test for HalfStudentT
* Tidy up distributions in the API documentation
    * Replace StudetTpos by HalfStudentT 
    * Zero-inflated Poisson was duplicate and ZeroInflatedNegativeBinomial was missing
    * Other missing distributions were Interpolated and MvStudentT

### CODE REVIEW

File: pymc3/distributions/continuous.py [hunk 1]
```diff
@@ -1534,12 +1533,79 @@ def _repr_latex_(self, name=None, dist=None):
                                                                 get_variable_name(beta))
 
 
-def StudentTpos(*args, **kwargs):
-    warnings.warn("StudentTpos has been deprecated. In future, use HalfStudentT instead.",
-                DeprecationWarning)
-    return HalfStudentT(*args, **kwargs)
+class HalfStudentT(Continuous):
```
Comment (twiecki) [inline pymc3/distributions/continuous.py]: Shouldn't we inherit from `PositiveContinuous` here?

Comment (aloctavodia) [reply pymc3/distributions/continuous.py]: you are right, I totally missed that!

Comment (aloctavodia) [reply pymc3/distributions/continuous.py]: fixed, thank you.

**Comment by twiecki** [comment]
Why can't we just do `Bound(StudentT, 0)`  (or whatever the syntax is), rather than reimplementing the logp?

**Comment by aloctavodia** [comment]
I started working on this a while ago and then I completely forget it until yesterday. I vaguely remember trying with `Bound` but I got into some trouble, I don't remember exactly what. Another reason (maybe not a very good one) is that implementing the logp is consistent with the implementation of `HalfNormal` and `HalfCauchy`. I think implementing the logp is not a bad idea, but if you want I can try using `Bound.`

**Comment by junpenglao** [comment]
If there is a speed advantage I think it is worth to implement it.

**Comment by aloctavodia** [comment]
I remember now! One of the problems I found is that a `Bound` distribution can not be used as a likelihood (or I did not find the proper way to do it). I think `Bound` is a nice tool for users, but probably distributions should be implemented explicitly.

**Comment by twiecki** [comment]
I see, yeah, that seems to be a short-coming of bound. Thanks!

**Comment by aloctavodia** [comment]
@twiecki thank you for the feedback :-) and also thanks to @junpenglao

**Comment by aseyboldt** [comment]
It seems this inadvertently changed the name of the scale parameter from `sd` to `beta`. This breaks some code. Could we just allow both?

**Comment by junpenglao** [comment]
I would say allow both or change it back, halfNormal is using sd no?

**Comment by aloctavodia** [comment]
I guess we should use `sd`, should we also use `sd` for HalfCauchy? (or maybe remove HalfCauchy...) I can take a look at this tomorrow, unless someone else wants to solve this.

**Comment by fonnesbeck** [comment]
The Student T uses sd (`lam`), so it would make sense to make it consistent. I would leave `HalfCauchy` as it is, unless there is a reason to remove it. Its a common choice for standard deviation  and GP lengthscale priors.

**Comment by twiecki** [comment]
`lam` is not the sd of the distribution, so we shouldn't use it for HalfCauchy or (Half)Student-T.

**Comment by fonnesbeck** [comment]
Sorry, right--It's the square-root precision, which is odd. In any case, I think it makes sense to have the half-* distributions scale parameters correspond to their full-support counterparts. I'd be in favor of changing `StudentT` to be a standard deviation, but it seems inconsistent to have them opposite.

**Comment by aloctavodia** [comment]
Just to be sure we are all on the same page:

For the StudentT distribution (as implemented)
* The parameter `sd` is not the standard deviation of the Student-T distribution (is just an scale parameter), although converges to the standard deviation as `nu` increases.
* The parameter `lam` is (1/sd)**0.5

So, to make things consistent the HalfStudentT should have both `sd` and `lam` parameters...

**Comment by fonnesbeck** [comment]
The way we have it implemented, it looks like `lam` is (1/sd). 

We also have this on line 939:

```python
assert_negative_support(lam, 'lam (sd)', 'StudentT')
```

**Comment by twiecki** [comment]
I don't think that's true: https://github.com/pymc-devs/pymc3/blob/master/pymc3/distributions/continuous.py#L935 The variance/sd rely on both `nu` and `lam`.

**Comment by aloctavodia** [comment]
I just run a couple of tests for the `StudentT` and  `lam` is effectively (1/sd)**0.5. I also checked empirically that `sd` converges to the standard deviation as nu increases. This is fine, because when nu is infinite you should recover the normal distribution, i.e the normal distribution is a limiting case of the StudentT.

**Comment by fonnesbeck** [comment]
Yes, I get the same. Seems odd to have a scale parameter that is sqrt(sqrt(1/var)), but there you go.

**Comment by aloctavodia** [comment]
Sorry guys I have been writing things backwards the correct expression is `sd` = (1/`lam`)**0.5 
and the computation of the variance here https://github.com/pymc-devs/pymc3/blob/master/pymc3/distributions/continuous.py#L935  is OK and from it we can see that as `nu` increases, the term  `(nu / (nu - 2))` approaches to 1 and then the variance approaches to (1/lam).

**Pull request merged**